src/prpy/util.py

```diff
@@ -749,3 +749,197 @@ def IsTimedTrajectory(trajectory):
     """
     cspec = trajectory.GetConfigurationSpecification()
     return cspec.ExtractDeltaTime(trajectory.GetWaypoint(0)) is not None
+
+
+def JointStatesFromTraj(robot, traj, times, derivatives=[0, 1, 2]):
+    """
+    Helper function to extract the joint position, velocity and acceleration
+    from an OpenRAVE trajectory.
+    @param robot The OpenRAVE robot
+    @param traj An OpenRAVE trajectory
+    @param times List of times in seconds
+    @param derivatives list of desired derivatives defaults to [0, 1, 2]
+    @return pva_list List of list of derivatives at specified times.
+                     Inserts 'None' for unavailable or undesired fields
+                     The i-th element is the i-th derivative of position
+                     Of size |times| x |derivatives|
+
+    """
+    duration = traj.GetDuration()
+
+    times = numpy.array(times)
+    if any(times > duration):
+        raise ValueError('Input times {0:} exceed duration {1:.2f}'
+                         .format(times, duration))
+
+    cspec = traj.GetConfigurationSpecification()
+    num_dofs = robot.GetDOF()
+    dof_indices = range(num_dofs)
+
+    pva_list = []
+    for t in times:
+        pva = [None] * len(derivatives)
+        trajdata = traj.Sample(t)
+        for i in range(len(derivatives)):
+            if i in derivatives:
+                pva[i] = cspec.ExtractJointValues(trajdata, robot,
+                                                  dof_indices, i)
+        pva_list.append(pva)
```

Comment:
This looks like a bug: both loops are indexed by `i`, so the outer loop does nothing. If we're sticking with the `list` API, then this should be:
```python
for i, deriv in enumerate(derivatives):
    pva[i] = cspec.ExtractJointValues(trajdata, robot, dof_indices, deriv)
```